### PR TITLE
Replace NotNullAttribute with thrown exceptions

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs [hunk 1]
```diff
@@ -160,8 +168,13 @@ public virtual ConflictResult Conflict()
         /// <param name="value">The content value to negotiate and format in the entity body.</param>
         /// <returns>A <see cref="NegotiatedContentResult{T}"/> with the specified values.</returns>
         [NonAction]
-        public virtual NegotiatedContentResult<T> Content<T>(HttpStatusCode statusCode, [NotNull] T value)
+        public virtual NegotiatedContentResult<T> Content<T>(HttpStatusCode statusCode, T value)
         {
+            if (value == null)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs]: This surprisingly doesn't cause a compilation issue and works at runtime for value types.

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs]: @pranavkm what does "works at runtime" mean? Does it throw if `value == default(T)` for example?

Not surprising there's no compilation warning. Compilers tend to be silent unless they're _sure_ you're doing something wrong.

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs]: This isn't surprising at all - how else would you write null-safe generic code?

**Comment by dougbu** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs]: @dougbu, it just no-ops for value types.

**Pull request closed without merging**